softlayer/vm/softlayer_virtual_guest_creator.go

```diff
@@ -36,17 +38,23 @@ func NewSoftLayerCreator(vmFinder Finder, softLayerClient sl.Client, agentOption
 		softLayerClient: softLayerClient,
 		agentOptions:    agentOptions,
 		logger:          logger,
+		featureOptions:  featureOptions,
 	}
 }
 
 func (c *softLayerVirtualGuestCreator) Create(agentID string, stemcell bslcstem.Stemcell, cloudProps VMCloudProperties, networks Networks, env Environment) (VM, error) {
 	for _, network := range networks {
 		switch network.Type {
 		case "dynamic":
-			if len(network.IP) == 0 {
+			if cloudProps.DisableOsReload || c.featureOptions.DisableOsReload {
```

Comment:
Why check both? Should it we just set `c.featureOptions.DisableOsReload` if `cloudProps.DisableOsReload`